lldb/source/Plugins/ObjectFile/ELF/ObjectFileELF.cpp

```diff
@@ -1284,6 +1285,44 @@ ObjectFileELF::RefineModuleDetailsFromNote(lldb_private::DataExtractor &data,
   return error;
 }
 
+void ObjectFileELF::ParseRISCVAttributes(DataExtractor &data, uint64_t length,
+                                         ArchSpec &arch_spec) {
+  lldb::offset_t Offset = 0;
+
+  uint8_t FormatVersion = data.GetU8(&Offset);
+  if (FormatVersion != llvm::ELFAttrs::Format_Version)
+    return;
+
+  Offset = Offset + sizeof(uint32_t); // Section Length
+  llvm::StringRef VendorName = data.GetCStr(&Offset);
+
+  if (VendorName != "riscv")
+    return;
+
+  llvm::StringRef attr = "";
+
+  while (Offset < length) {
+    uint8_t Tag = data.GetU8(&Offset);
+    uint32_t Size = data.GetU32(&Offset);
+
+    if (Tag != llvm::ELFAttrs::File || Size == 0)
+      continue;
+
+    while (Offset < length) {
+      uint64_t Tag = data.GetULEB128(&Offset);
+      if (Tag == llvm::RISCVAttrs::ARCH) {
+        attr = data.GetCStr(&Offset);
+        break;
+      } else {
+        data.GetULEB128(&Offset);
+      }
+    }
+  }
+  if (!attr.empty() && attr.contains("xqci")) {
+    arch_spec.SetAdditionalDisassemblyFeatureStr("+xqci,");
```

Comment:
Isn't this loop only going to add `+xcqi` to the ArchSpec disassembly features str if the ELF file has the feature `xcqi`, else it does nothing?  Is that the only feature that can be specified in the ELF file attribute?